gridstatus/spp.py

```diff
@@ -1216,6 +1218,109 @@ def _match(
             )
         ]
 
+    @support_date_range("DAY_START")
+    def get_hourly_load(self, date, end=None, verbose=False):
+        """Get Hourly Load
+
+        Supports recent data. For historical annual data use get_hourly_load_annual
+
+        Args:
+            date: start date
+            end: end date
+
+        Returns:
+            pd.DataFrame: Hourly Load
+        """
+        if date in ["today", "latest"] or utils.is_today(
+            date, tz=self.default_timezone
+        ):
+            raise NotSupported("Only historical data is available for hourly load data")
+
+        url = f"{FILE_BROWSER_DOWNLOAD_URL}/hourly-load?path=/{date.strftime('%Y')}/DAILY_HOURLY_LOAD-{date.strftime('%Y%m%d')}.csv"  # noqa
+        msg = f"Downloading {url}"
+        log(msg, verbose)
+        df = pd.read_csv(url)
+        df = df.rename(columns=lambda x: x.strip())
```

Comment:
i think the goal is to have `_process_hourly_load` do everything besides download the data, so i think move this line in there.

Reply:
Sounds good. I'll see if I can also move some of the clean up logic from `get_hourly_load_annual` into `_process_hourly_load` as well.